components/Weather/index.js

```diff
@@ -0,0 +1,76 @@
+import axios from "axios";
+import { useState, useEffect } from "react";
+import styled from "styled-components";
+import Icon from "../Icons";
+
+export default function Weather({ outivity }) {
+  const [weatherData, setWeatherData] = useState(null);
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  const fetchData = async () => {
+    try {
+      const response = await axios.get(
+        `https://api.openweathermap.org/data/2.5/weather?lat=${outivity.lat}&lon=${outivity.lng}&units=metric&appid=9e7a02e3cc30d92b7c137c9919b52688`
+      );
+
+      setWeatherData(response.data);
+      setErrorMessage(null);
+    } catch (error) {
+      console.error("Error fetching weather data: ", error);
+      setErrorMessage("Weather data is not available at the moment...");
+    }
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, [outivity.lat, outivity.lng]);
```

Comment:
The dependency array normally determines a usestate and the useEffect is run each time this state changes. I am not seeing a case, where a lat or a lng of an outivity can change.

Reply:
Thank you, we have changed it :)